cmd/main_test.go

```diff
@@ -250,46 +251,17 @@ var _ = DescribeTable("Specific workspace endpoint", func(endpoint string, heade
 		`{"message":"Not Found"}`),
 )
 
-func CreateKubeconfigFileForRestConfig(restConfig rest.Config) string {
-	clusters := make(map[string]*clientcmdapi.Cluster)
-	clusters["default-cluster"] = &clientcmdapi.Cluster{
-		Server:                   restConfig.Host,
-		CertificateAuthorityData: restConfig.CAData,
-	}
-	contexts := make(map[string]*clientcmdapi.Context)
-	contexts["default-context"] = &clientcmdapi.Context{
-		Cluster:  "default-cluster",
-		AuthInfo: "default-user",
-	}
-	authinfos := make(map[string]*clientcmdapi.AuthInfo)
-	authinfos["default-user"] = &clientcmdapi.AuthInfo{
-		ClientCertificateData: restConfig.CertData,
-		ClientKeyData:         restConfig.KeyData,
-	}
-	clientConfig := clientcmdapi.Config{
-		Kind:           "Config",
-		APIVersion:     "v1",
-		Clusters:       clusters,
-		Contexts:       contexts,
-		CurrentContext: "default-context",
-		AuthInfos:      authinfos,
-	}
-	kubeConfigFile, _ := os.CreateTemp("", "kubeconfig")
-	_ = clientcmd.WriteToFile(clientConfig, kubeConfigFile.Name())
-	return kubeConfigFile.Name()
-}
-
 var serverProcess *exec.Cmd
+var serverCancelFunc context.CancelFunc
 
 var _ = BeforeSuite(func() {
-	testEnv = &envtest.Environment{}
-	cfg, err := testEnv.Start()
-	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error creating the envtest environment during test setup: %v", err))
-	kubeconfigPath := CreateKubeconfigFileForRestConfig(*cfg)
-	os.Setenv("KUBECONFIG", kubeconfigPath)
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme})
-	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error creating the client during test setup: %v", err))
-	Expect(k8sClient).NotTo(BeNil())
+	schema := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(schema))
+	testEnv = &envtest.Environment{BinaryAssetsDirectory: "../bin/k8s/1.29.0-linux-amd64/"}
```

Comment:
How do you see us keeping track of the version of this binary directory?

Reply:
it's created when you run `make envtest`. The versions of the dependencies for build/test exist in the Makefile (it was generated by kubebuilder).